x-pack/legacy/plugins/siem/public/containers/detection_engine/signals/errors_types/get_index_error.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { MessageBody } from '../../../../components/ml/api/throw_if_not_ok';
+
+export class SignalIndexError extends Error {
+  message: string = '';
+  statusCode: number = -1;
+  error: string = '';
+
+  constructor(errObj: MessageBody) {
+    super(errObj.message);
+    this.message = errObj.message ?? '';
+    this.statusCode = errObj.statusCode ?? -1;
+    this.error = errObj.error ?? '';
```

Comment:
nit: Instead of duplicating the default values you can reference the initial super state

``` ts
this.message = errObj.message ?? this.message;
this.statusCode = errObj.statusCode ?? this.statusCode;
this.error = errObj.error ?? this.error;
```

May want to set a custom name as well too

``` ts
this.name = 'SignalIndexError';
```

so it'll print as:

``` ts
SignalIndexError: faux message
    at eval (eval at <anonymous> (https://www.typescriptlang.org/play/main-3.js:1239:7), <anonymous>:17:13)
    at https://www.typescriptlang.org/play/main-3.js:1239:7
```